cmd/kube-events-exporter/main.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2020 Red Hat, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rhobs/kube-events-exporter/internal/exporter"
+	"github.com/rhobs/kube-events-exporter/internal/http"
+	"github.com/rhobs/kube-events-exporter/internal/options"
+	"github.com/rhobs/kube-events-exporter/internal/version"
+
+	"k8s.io/klog"
+)
+
+func main() {
+	opts := options.NewOptions()
+	opts.AddFlags()
+
+	err := opts.Parse()
+	if err != nil {
+		klog.Fatalf("Error: %s", err)
+	}
+
+	if opts.Version {
+		fmt.Printf("%#v\n", version.GetVersion())
+		os.Exit(0)
+	}
+
+	registry := prometheus.NewRegistry()
+	exporterRegistry := exporter.NewExporterRegistry()
+
+	// Serve metrics about the exporter.
+	go http.ServeExporterMetrics(opts.ExporterHost, opts.ExporterPort, exporterRegistry)
```

Comment:
Not strictly necessary but you might want to look into run.Group eventually to orchestrate the boot and shutdown of multiple long running go routines.